bin/libs/component.sh

```diff
@@ -501,66 +513,112 @@ zis_plugin_install() {
+          return 201
         fi
         while read samplib_key_value; do
           prefix=$(echo "$samplib_key_value" | cut -c -2)
           if [ "$prefix" = "//" ] || [ "$prefix" = "* " ] || [ "$prefix" = "" ]; then
             continue
           fi
-          grep -x "$samplib_key_value" "$parmlib_member_as_unix_file" > /dev/null
+          print_trace "$(grep -x "$samplib_key_value" "$parmlib_member_as_unix_file")"
           if [ $? -eq 0 ]; then
-            echo "The key-value pair $samplib_key_value is being skipped because it's already there and hasn't changed."
+            print_message "The key-value pair $samplib_key_value is being skipped because it's already there and hasn't changed."
             continue
           else
-            samplib_key=$(get_key_of_string "$samplib_key_value")
-            if [ "$samplib_key" = "" ]; then
-              exit 3
-            fi
-            # In the case of a key not being there, an empty string will be returned.
-            num=$(get_line_number_of_key "$samplib_key" "$parmlib_member_as_unix_file")
-            if [ "$num" != "" ]; then
-              parsed_zwes_zis_parmlib_keys=$(replace "${zwes_zis_parmlib_keys}" "." "_") # replace . with _ in keyname for working key search
-              parsed_samplib_key=$(replace "${samplib_key}" "." "_") # replace . with _ in keyname for working key search
-              samplib_key_value=$(read_json_string ${parsed_zwes_zis_parmlib_keys} ".${parsed_samplib_key}")
-              if [ "$samplib_key_value" = "list" ]; then
-              # The key is comma separated list
-                parmlib_key_value=$(get_string_at_line_number "$num" "$parmlib_member_as_unix_file")
-                parmlib_value=$(get_value_of_string "$parmlib_key_value")
-                samplib_value=$(get_value_of_string "$samplib_key_value")
-                is_substr_of "$samplib_value" "$parmlib_value"
-                if [ $? -eq 0 ]; then
-                  new_parmlib_key_value="$samplib_key=$parmlib_value,$samplib_value"
-                  remove_key_value_at_line_number $num "$parmlib_member_as_unix_file"
-                  add_key_value_at_end_of_file "$new_parmlib_key_value" "$parmlib_member_as_unix_file"
-                  changed=1
-                fi
-              else
-                # The key is not special and the value is different.
-                remove_key_value_at_line_number "$num" "$parmlib_member_as_unix_file"
-                add_key_value_at_end_of_file "$samplib_key_value" "$parmlib_member_as_unix_file"
-                changed=1
-              fi
+            update_uss_parmlib_key_value "$samplib_key_value" "$parmlib_member_as_unix_file"
+            if [ $? -ne 0 ]; then
+              print_message "Failed to install ZIS plugin: ${zis_plugin_id}"
+              exit 1
             else
-              # The key doesn't exist. Just add the key-value pair to the end of the file.
```

Comment:
So this ends up being
else
fi
which is kind of vacuous. You should remove the else.